stainless_extraction/src/ty.rs

```diff
@@ -329,6 +327,19 @@ impl<'l, 'tcx> BaseExtractor<'l, 'tcx> {
     }
   }
 
+  fn all_predicates_of(&self, def_id: DefId) -> Vec<&(Predicate<'tcx>, Span)> {
+    let predicates = self.tcx.predicates_defined_on(def_id);
+    let mut all_predicates: Vec<GenericPredicates> = vec![predicates];
```

Comment:
Doesn't seem like the type annotation is needed here?

```suggestion
    let mut all_predicates = vec![predicates];
```